.github/workflows/pipeline.yml

```diff
@@ -63,57 +75,99 @@ jobs:
           yarn build
           yarn test
         working-directory: ./test/bucklescript
-      
-      - name: Get Esy binary path
-        id: esy_binary_path
-        shell: bash
-        run: |
-          # from https://stackoverflow.com/a/24848739/617787
-          s=$(pwd); d=$(dirname $(esy x which tailwind-ppx));
-          while [ "${d#$s/}" == "${d}" ]
-          do s=$(dirname $s);b="../${b}"; done;
-          if [ "${{ matrix.os }}" == "windows-latest" ]; then
-            ESY__BINARY_PATH=${b}${d#$s/}/tailwind-ppx.exe
-          else
-            ESY__BINARY_PATH=${b}${d#$s/}/tailwind-ppx
-          fi
-          echo "ESY__BINARY_PATH: $ESY__BINARY_PATH"
-          echo "##[set-output name=path;]$ESY__BINARY_PATH"
-
-      - name: Upload artifacts for ${{ matrix.os }}
-        if: github.event_name != 'pull_request' && startsWith(github.ref, 'refs/tags/')
-        uses: actions/upload-artifact@master
+
+      # Create the NPM package for this platform
+      - name: Create npm package
+        run: esy npm-release
+      - uses: actions/upload-artifact@v2
         with:
           name: ${{ matrix.os }}
-          path: ${{ steps.esy_binary_path.outputs.path }}
+          path: _release/
 
+      # Cleanup build cache if dependencies have changed
+      - name: Clean build cache
+        if: steps.deps-cache.outputs.cache-hit != 'true'
+        run: esy cleanup .
 
-      - name: Get Esy UsePpx binary path
-        id: esy_use_ppx_binary_path
-        shell: bash
+  prepare-publish:
+    name: Prepare publish to npm
+    needs: build_and_test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+          ref: ${{ github.ref }}
+          submodules: true
```

Comment:
Can you point me to some docs or explain what each of these options under the `with` field do?

Reply:
Indeed it's not necessary I guess so I removed them :)